lib/net/peer.js

```diff
@@ -1900,6 +1903,43 @@ class Peer extends EventEmitter {
     return false;
   }
 
+  /**
+   * Increase dosscore on peer.
+   * @param {Number} dosScore
+   * @param {Number} time
+   * @returns {Boolean}
+   */
+
+  increaseDos(dosScore, time) {
+    // dosScore is a time-bound transient banscore useful to detect DOS
+    // attacks by repeated requests for a message (e.g. getproof).  Calling
+    // increaseBan with a dosScore multiple times will increase dosScore in a
+    // logarithmically decaying fashion wrt time between requests to penalize
+    // short bursts of repeated requests.  The dosScore decays back to 0 in
+    // 30mins, i.e. 1800s.
+    const seconds = (time - this.lastDos) / 1000;
+    if (seconds > common.DOS_RESET_TIME) {
+      this.dosScore = 0;
+    } else if (this.dosScore > 1 && seconds > 0) {
+      if (seconds < 64) {
+        this.dosScore *= common.DOS_DECAY_FACTOR[Math.floor(seconds)];
+      } else {
+        this.dosScore *= Math.exp(-1.0 * seconds * (Math.log(2) / 60));
+      }
+    }
+    this.lastDos = time;
+
+    this.dosScore = Math.floor(this.dosScore) + dosScore;
+
+    if (this.dosScore >= this.options.dosScore) {
+      this.logger.debug('Ban threshold exceeded (%s).', this.hostname());
```

Comment:
Could you update this message so we know it was caused by the `dosScore`?

Reply:
Fixed.